src/tooling/piral-cli/src/common/scripts.ts

```diff
@@ -36,7 +36,23 @@ export function runScript(script: string, cwd = process.cwd(), output: NodeJS.Wr
 
 export function runCommand(exe: string, args: Array<string>, cwd: string, output?: NodeJS.WritableStream) {
   const npmCommand = isWindows ? `${exe}.cmd` : exe;
-  const cmd = [npmCommand, ...args].join(' ');
+  const sanitizedArgs = sanitizeCmdArgs(args);
+  const cmd = [npmCommand, ...sanitizedArgs].join(' ');
   log('generalDebug_0003', `Applying cmd "${cmd}" in directory "${cwd}".`);
   return runScript(cmd, cwd, output);
 }
+
+function sanitizeCmdArgs(args: Array<string>) {
+  // Introduced for fixing https://github.com/smapiot/piral/issues/259.
+  // If an arg contains a whitespace, it can be incorrectly interpreted as two separate arguments.
+  // For the moment, it's fixed by simply wrapping each arg in "quotation marks".
+  // To ensure that this doesn't conflict with user-provided quotation marks, *one* leading/trailing " ' char
+  // is trimmed from each arg. Only one, because multiple are usually deliberate and removing them thus becomes
+  // a destructive action (e.g. we wouldn't want to replace "'foo'" with "foo", because one pair of quotes disappears here).
+  return args.map(arg => {
+    const trimWhiteSpaceAndQuotMarks = /^\s*['"]?(.*?)['"]?\s*$/;
```

Comment:
I would argue that our algorithm is better -> because if a user of `runCommand` already decided to put in `'` or `"` then something must be known here. I would then leave it as-is.

Also I am unsure if `npm "publish"` looks / works as much as `npm publish`. Can you *always* just quote? In any case I would only quote if we find a space -> less chance to destroy anything.

Note that there are different kinds of quotes on nix systems - and it makes a difference if its single or double quoted.

https://stackoverflow.com/questions/6697753/difference-between-single-and-double-quotes-in-bash

Also characters like `@` and `*` have a special meaning when in double quotes on Bash. Therefore scaffolding from `pilet new sample-piral@next` will potentially stop working with this approach.

I think the spaces approach *is* better because right now the only potential case for spaces is a file path - here it makes sense to quote (potentially even use double quotes on Windows and single quotes on nix systems).

Reply:
Put like this, it makes a lot of sense - I'll get right on it!

When there are whitesapces, is the approach of only trimming one quote okay or would you remove every leading/trailing one (e.g. `"'""abc` -> `abc`)?

Reply:
Good question. I actually don't know. I think if there is a leading quote we should assume the input has to remain as-is.